src/main/java/com/eternalcode/parcellockers/ParcelLockers.java

```diff
@@ -74,27 +81,41 @@ public void onEnable() {
 
         HikariDataSource dataSource = DataSourceFactory.buildHikariDataSource(config, this.getDataFolder());
 
-        ParcelLockerRepository parcelLockerRepository = new ParcelLockerDatabaseService(dataSource);
+        ParcelLockerDatabaseService parcelLockerDatabaseService = new ParcelLockerDatabaseService(dataSource);
+        parcelLockerDatabaseService.updatePositionCache();
+        
         ParcelDatabaseService parcelRepository = new ParcelDatabaseService(dataSource);
 
-        ParcelManager parcelManager = new ParcelManager(this, config, announcer, parcelRepository, parcelLockerRepository);
-        ParcelLockerManager parcelLockerManager = new ParcelLockerManager(parcelLockerRepository);
-
-        MainGUI mainGUI = new MainGUI(this, this.getServer(), miniMessage, config, parcelRepository, parcelLockerRepository);
-        ParcelListGUI parcelListGUI = new ParcelListGUI(this, this.getServer(), miniMessage, config, parcelRepository, parcelLockerRepository, mainGUI);
+        ParcelManager parcelManager = new ParcelManager(this, config, announcer, parcelRepository, parcelLockerDatabaseService);
+        ParcelLockerManager parcelLockerManager = new ParcelLockerManager(parcelLockerDatabaseService);
 
+        MainGUI mainGUI = new MainGUI(this, this.getServer(), miniMessage, config, parcelRepository, parcelLockerDatabaseService);
```

Comment:
Don't use **this.getServer()**.
Store local variable `Server server = this.getServer();`
And pass it on like this:
```suggestion
        MainGUI mainGUI = new MainGUI(this, server, miniMessage, config, parcelRepository, parcelLockerDatabaseService);
```